src/pipeline_migration/actions/add_task.py

```diff
@@ -0,0 +1,406 @@
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    add_task_parser.add_argument(
+        "task",
+        help="Konflux task name to add. This is the actual task name defined in "
+        "konflux-ci/build-definitions. If a trusted artifact task is being added, "
+        "ensure the task name has suffix -oci-ta, e.g. buildah-oci-ta.",
+    )
+    add_task_parser.add_argument(
+        "file_or_dir",
+        nargs="*",
+        help="Specify pipeline files to where the task is added. "
+        "A pipeline can be included in a PipelineRun definition or a single Pipeline itself. "
+        "Directory path can also be specified alternatively. %(prog)s searches pipelines "
+        "from all YAML files from the first level of the directory.",
+    )
+    add_task_parser.add_argument(
+        "-n",
+        "--pipeline-task-name",
+        metavar="NAME",
+        help="Task name used in pipeline to configure the added task, in case it should be "
+        "different from the task name inside the bundle. For example, build task has name "
+        "build-container in the pipeline, but the actual task name is buildah inside bundle. "
+        "This is also useful for adding trusted artifact tasks, e.g. name the task as "
+        "sast-coverity-check but the actual task name is sast-coverity-check-oci-ta.",
+    )
+    add_task_parser.add_argument(
+        "-a",
+        "--run-after",
+        metavar="TASK_NAME",
+        dest="run_after",
+        action="append",
+        help="Name of task running before the adding task. "
+        "This is the task name used in the build pipeline definition. "
+        "This argument can be specified multiple times to add more than one tasks.",
+    )
+    add_task_parser.add_argument(
+        "-p",
+        "--param",
+        type=task_param,
+        metavar="PARAM",
+        dest="params",
+        action="append",
+        help="Specify a task parameter that consists of comma-separated name and value. "
+        "This argument can be specified multiple times to add more than one parameters.",
+    )
+    add_task_parser.add_argument(
+        "-s",
+        "--skip-checks",
+        action="store_true",
+        dest="skip_checks",
+        help="Skip this task if it can be skipped as a check for a fast build.",
+    )
+    add_task_parser.add_argument(
+        "-r",
+        "--bundle-ref",
+        type=konflux_task_bundle_reference,
+        metavar="IMAGE_REF",
+        dest="bundle_ref",
+        help="Use Tekton bundle resolver to reference the expected task bundle. "
+        "The full reference has to include both tag and digest, "
+        "e.g. registry/org/task-name:tag@digest."
+        "If omitted, the latest task bundle is queried from the registry.",
+    )
+    add_task_parser.add_argument(
+        "-g",
+        "--git-add",
+        dest="git_add",
+        action="store_true",
+        help="Add the modified files to git index.",
+    )
```

Comment:
This seems a bit unnecessary, what is the reason for having this option?

Reply:
Just wanting to provide a convenient way to to add to git index if multiple files and directories across repositories are specified through command line.

Reply:
Hmm, I see. Does it simplify the workflow much? The user would still have to go to each repo individually, commit the changes, push them and open a PR

Reply:
> Does it simplify the workflow much?

I don't have a good answer for now :) We'll see.

Reply:
Ok, we can keep it and see